packages/blade/src/components/Button/BaseButton/__tests__/BaseButton.web.test.tsx

```diff
@@ -0,0 +1,113 @@
+    try {
+      // @ts-expect-error testing failure case when there is no icon or text passed
+      renderWithTheme(<BaseButton />);
+    } catch (error: unknown) {
+      if (error instanceof Error) {
+        expect(error.message).toEqual(
+          `[Blade: BaseButton]: Cannot render a BaseButton without an icon or text`,
+        );
+      }
+    }
+  });
+
+  it('should render xsmall size button', () => {
+    const buttonText = 'Pay Now';
+    const { container } = renderWithTheme(<BaseButton size="xsmall">{buttonText}</BaseButton>);
+    expect(container).toMatchSnapshot();
+  });
+  it('should render small size button', () => {
+    const buttonText = 'Pay Now';
+    const { container } = renderWithTheme(<BaseButton size="small">{buttonText}</BaseButton>);
+    expect(container).toMatchSnapshot();
+  });
+  it('should render medium size button', () => {
+    const buttonText = 'Pay Now';
+    const { container } = renderWithTheme(<BaseButton size="medium">{buttonText}</BaseButton>);
+    expect(container).toMatchSnapshot();
+  });
+  it('should render large size button', () => {
+    const buttonText = 'Pay Now';
+    const { container } = renderWithTheme(<BaseButton size="large">{buttonText}</BaseButton>);
+    expect(container).toMatchSnapshot();
+  });
+  it('should render button with icon without text', () => {
+    const { container } = renderWithTheme(<BaseButton icon={CreditCardIcon} />);
+    expect(container).toMatchSnapshot();
+  });
+  it('should render button with icon with default iconPosition', () => {
+    const buttonText = 'Pay Now';
+    const { container } = renderWithTheme(
+      <BaseButton icon={CreditCardIcon}>{buttonText}</BaseButton>,
+    );
+    expect(container).toMatchSnapshot();
+  });
+  it('should render button with icon with left iconPosition', () => {
+    const buttonText = 'Pay Now';
+    const { container } = renderWithTheme(
+      <BaseButton iconPosition="left" icon={CreditCardIcon}>
+        {buttonText}
+      </BaseButton>,
+    );
+    expect(container).toMatchSnapshot();
+  });
+  it('should render button with icon with right iconPosition', () => {
+    const buttonText = 'Pay Now';
+    const { container } = renderWithTheme(
+      <BaseButton iconPosition="right" icon={CreditCardIcon}>
+        {buttonText}
+      </BaseButton>,
+    );
+    expect(container).toMatchSnapshot();
+  });
+  it('should render button with full width', () => {
+    const buttonText = 'Pay Now';
+    const { container } = renderWithTheme(<BaseButton isFullWidth={true}>{buttonText}</BaseButton>);
+    expect(container).toMatchSnapshot();
+  });
+  it('should render disabled button', () => {
+    const buttonText = 'Pay Now';
+    const { container, getByRole } = renderWithTheme(
+      <BaseButton isDisabled={true}>{buttonText}</BaseButton>,
+    );
+    expect(container).toMatchSnapshot();
+    expect(getByRole('button')).toBeDisabled();
```

Comment:
we don't have similar thing on native? over there we have to simulate that with fireEvent.

Reply:
Don't have the similar equivalent there. We could do `container.props.disabled` assertion but its as good as doing a snapshot check which will check for prop as well as styles